### PR TITLE
IRO-1666 - Add basic tests to UI Kit

### PR DESCRIPTION
**Pull request by brekk**
Updated `package-scripts.js`:

`nps test` - Run tests
`nps test.watch` - Run tests and re-run on changes
`nps test.snapshot` - Run tests and automatically update snapshots

Added `__tests__`:

- `simple.tsx` - basic proof of concept. Lots of time fighting with yarn resolutions to get the right versions of the dependency tree loaded _without_ breaking the storybook build.

Added `tsconfig.test.json` for specific TS config while testing.

Updated the `scripts/force-static-versions.js` file to deal with `peerDependencies` as well.

### CODE REVIEW

File: .storybook/main.js [hunk 1]
```diff
@@ -22,17 +20,17 @@ module.exports = {
...
-    };
-  }
-}
\ No newline at end of file
+        plugins: [...(resolve.plugins || []), new TsconfigPathsPlugin()],
+      },
+    }
+  },
+  features: {
+    emotionAlias: false,
```
Comment (azhurauski-sc) [inline .storybook/main.js:34]: 👍

Comment (brekk) [reply .storybook/main.js:34]: Haha yeah it took a long time to find that!

File: jest.utils.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  return (
+    <ThemeProvider theme={extendTheme(IronFishTheme)}>{children}</ThemeProvider>
+  )
+}
+
+export const render = (C: any, options = { theme: IronFishTheme }) =>
+  defaultRender(C, {
+    wrapper: ChakraRenderer,
+    ...options,
+  })
```
Comment (brekk) [inline jest.utils.tsx:60]: This is now a reusable renderer for Chakra components.

File: jest.utils.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
+import { ReactType, createElement } from 'react'
+import '@testing-library/jest-dom'
+import * as React from 'react'
+import { ChakraProvider as ThemeProvider, extendTheme } from '@chakra-ui/react'
+import { render as defaultRender } from '@testing-library/react'
+
+import { IronFishTheme } from 'theme/theme'
+
+import { NextRouter } from 'next/router'
```
Comment (ndemidovich-sc) [inline jest.utils.tsx:9]: Can we avoid of usage nextjs packages?

File: jest.config.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+module.exports = {
+  automock: false,
+  transform: {
+    '^.+\\.(js|ts|tsx)$': 'ts-jest',
+  },
+  // setupFilesAfterEnv: ['<rootDir>/jest.setup.ts'],
+  // https://github.com/testing-library/react-hooks-testing-library/issues/294
+  moduleDirectories: ['node_modules', '<rootDir>/node_modules', '.'],
+  moduleFileExtensions: ['ts', 'tsx', 'js', 'json', 'jsx', 'json'],
```
Comment (ndemidovich-sc) [inline jest.config.js:25]: Remove duplication of `json`

File: tsconfig.test.json [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    "baseUrl": ".",
+    "target": "es5",
+    "lib": ["dom", "dom.iterable", "esnext"],
+    "allowJs": true,
+    "skipLibCheck": true,
+    "strict": true,
+    "forceConsistentCasingInFileNames": true,
+    "noEmit": true,
+    "esModuleInterop": true,
+    "module": "esnext",
```
Comment (ndemidovich-sc) [inline tsconfig.test.json:12]: Can we avoid of using Nextjs?

**Review by ndemidovich-sc — CHANGES_REQUESTED** [review]

File: tsconfig.test.json (hunk 5 above)
Comment (brekk) [reply tsconfig.test.json:12]: In general yes. However this line is related to `esnext`, which is a JS standard, not a package related to next.

**Review by ndemidovich-sc — APPROVED** [review]

**Pull request merged**